.github/workflows/cachegrind.yml

```diff
@@ -0,0 +1,47 @@
+name: Valgrind Cachegrind
+
+on:
+  pull_request:
+    branches: [main]
+jobs:
+  valgrind:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup | Rust
+        uses: ATiltedTree/setup-rust@v1
+        with:
+          rust-version: stable
+
+      - name: Install Valgrind
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y valgrind
+
+      - name: Install cg_annotate
+        run: |
+          sudo apt-get install -y kcachegrind
+
+      - name: Run Valgrind
+        run: |
+          cargo b -r -p revm-test --bin snailtracer
+          valgrind --tool=cachegrind target/release/snailtracer
+
+          # Extract relevant information from cachegrind.out file
+          cg_annotate cachegrind.out.* > cachegrind_results.txt
```

Comment:
We don't need this file but just the output of the Valgrind.

Something like this:
`valgrind --tool=cachegrind target/release/snailtracer 2>&1 | tail -n 1 | tee cachegrind_results.txt`

Otherwise it looks nice

Reply:
why just the first line?

Reply:
It is more resonable then 64k and most important, but i like what you did in the end.

Reply:
it was just the first 64k characters not lines! 

looks less spammy :P